File: aula_02/Personagem.java
public class Personagem {
    private String nome;
    private int vida;
    private int forca;
    public Personagem(String nome, int vida, int forca){
        this.nome = nome;
        this.vida = vida;
        this.forca = forca;
    }
    public void receber_dano(int dano){
        this.vida -= dano;
        if(this.vida < 0)this.vida = 0;
    }
    public boolean esta_vivo(){
        return this.vida > 0;
    }
    public void ficha(){
        System.out.println("Nome: " + this.nome);
        System.out.println("Vida: " + this.vida);
        System.out.println("Forca: " + this.forca);
    }
    public void atacar(Personagem alvo){
        alvo.receber_dano(this.forca);
    }
    public static void main(String[] args){
        Personagem heroi = new Personagem("Heroi", 100, 10);
        Personagem chefe = new Personagem("Chefe",  150, 15);
        System.out.println("\n----Ficha de Personagem----\n");
        heroi.ficha();
        System.out.println();
        chefe.ficha();
        while (heroi.esta_vivo() && chefe.esta_vivo()){
            heroi.atacar(chefe);
            if (chefe.esta_vivo()){
                chefe.atacar(heroi);
            }
        }
        System.out.println("----Apos a luta----\n");
        heroi.ficha();
        System.out.println();
        chefe.ficha();
        if (heroi.esta_vivo()){
            System.out.println("Vitoria do heroi\n");
        }else{
            System.out.println("Vitoria do chefe");
        }
    }
}

File: aula_02/personagem.py
class Personagem:
    def __init__(self, nome: str, vida: int, forca: int):
        self.__nome = nome
        self.__vida = vida
        self.__forca = forca
    def receber_dano(self, dano: int):
        self.__vida -= dano
        if self.__vida < 0:
            self.__vida = 0
    def esta_vivo(self) -> bool:
        return self.__vida > 0
    def ficha(self):
        print(f"Nome: {self.__nome}")
        print(f"Vida: {self.__vida}")
        print(f"Forca: {self.__forca}")
    def atacar(self, alvo: 'Personagem'):
        alvo.receber_dano(self.__forca)
    @property
    def nome(self):
        return self.__nome
    @property
    def vida(self):
        return self.__vida
    @property
    def forca(self):
        return self.__forca
if __name__ == "__main__":
    heroi = Personagem("Heroi", 100, 10)
    chefe = Personagem("Chefe", 150, 15)
    print("\n----Ficha de Personagem----\n")
    heroi.ficha()
    print()
    chefe.ficha()
    while heroi.esta_vivo() and chefe.esta_vivo():
        heroi.atacar(chefe)
        if chefe.esta_vivo():
            chefe.atacar(heroi)
    print("\n----Apos a luta----\n")
    heroi.ficha()
    print()
    chefe.ficha()
    if heroi.esta_vivo():
        print("Vitoria do heroi\n")
    else:
        print("Vitoria do chefe")
